src/refract/RenderJSONVisitor.cc

```diff
@@ -89,6 +89,33 @@ namespace refract
 
         return false;
     }
+    
+    template<typename T>
+    bool isOptional(const T& e) {
```

Comment:
This function is just duplicate of `isNullable`. Please optimise it by making a single function named `isTypeAttribute` taking a single argument.